src/Uno.UI/UI/Xaml/VisualStateGroup.cs

```diff
@@ -303,19 +303,10 @@ void ApplyTargetState()
 				// Starts target state animation
 				if (target.animation is { } stateAnimation)
 				{
-					stateAnimation.Completed += OnStateStoryboardCompleted;
 					stateAnimation.Begin();
-
-					void OnStateStoryboardCompleted(object s, object a)
-					{
-						state.Storyboard.Completed -= OnStateStoryboardCompleted;
-						onStateChanged();
-					}
-				}
-				else
-				{
-					onStateChanged();
```

Comment:
/cc @dr1rrb does this relate to changes you're making ?

Reply:
This feels odd removing this part is not breaking anything else, maybe david will know more.

Reply:
@jeromelaban Yeah I have the same feeling. But I saw on UWP even for non-Forever animations, the state changed is fired much earlier than Storyboard's Completed. So it seems to make some sense, but I don't know why this code existed at the first place.

Reply:
@dr1rrb Would you be able to check? Thanks!

Reply:
I think this was made when the "transitions" concept was not applied properly, but I've tested and I confirm that the `CurrentStateChanged` event is raised **after** the transition but at the beginning of the state animation.